realm/realm-library/src/objectServer/java/io/realm/SyncUser.java

```diff
@@ -274,7 +275,31 @@ public void logout() {
             // the similar SyncConfiguration using the same identity, but with different (new)
             // refresh-token.
             realms.clear();
-            SyncManager.notifyUserLoggedOut(SyncUser.this);
+
+            // Finally revoke server token. The local user is logged out in any case.
+            final AuthenticationServer server = SyncManager.getAuthServer();
+            // don't reference directly the refreshToken inside the revoke request
+            // as it may revoke the newly acquired and refresh_token
+            final Token refreshTokenToBeRevoked = refreshToken;
+
+            ThreadPoolExecutor networkPoolExecutor = SyncManager.NETWORK_POOL_EXECUTOR;
+            networkPoolExecutor.submit(new ExponentialBackoffTask<LogoutResponse>() {
+
+                @Override
+                protected LogoutResponse execute() {
+                    return server.logout(refreshTokenToBeRevoked, getAuthenticationUrl());
+                }
+
+                @Override
+                protected void onSuccess(LogoutResponse response) {
```

Comment:
So the ROS solved the logout issue?

It is not related with this PR, but shall we just ignore the response of logout and just notify user that it has been logged out? Could user logout again if the first try failed? I guess user cannot ?

Reply:
Yes, you can now revoke tokens.

Right now the `AuthenticationListener` was supposed to represent the server state, so it shouldn't be triggered for only a local logout (where you just delete the refreshToken).

But that said, there are a lot of cases where these semantics get a bit murky, e.g. it isn't triggered if another device logs you out. Only way we detect this is by having something throw an `invalid access` error somewhere.

The biggest problem as you point out is that if the revoke call fails, the user is not notified, but they can call `logout` again. Only access tokens to Realms are removed and they are marked as logged out, but they keep the refresh token.

I'll create an issue to track this as it probably requires some discussion

Reply:
https://github.com/realm/realm-java/issues/5288